### DISCUSSION TITLE
Next.js API routes + Node-APN: Can't get path to .p8 Apple key file to send push notification after deploying to Vercel

### DISCUSSION THREAD

**Discussion by oskargaw** [vercel/vercel#6507] [Help] [answered]
Hi! 
In my Next.js app, I would like to be able to get iOS device tokens from a request body and send notifications to these tokens whenever someone hits the endpoint that I've created in my `/pages/api/sendPushNotification.ts`. I'm using `node-apn` package (https://www.npmjs.com/package/apn) to establish a connection between my Node.js server code (located in `pages/api/sendPushNotification.ts`) and Apple Push Notification service. I already have my `.p8 key file` and I know that it probably shouldn't be commited, but I couldn't figure out any other way to pass a path to it  🤷🏻‍♂️. I have it located in `/public/apn/AuthKey.p8`. Everything works fine locally, when I pass my local absolute path to `.p8 key file` into this config from `node-apn` package:
```
var options = {
  token: {
    key: "path/to/APNsAuthKey_XXXXXXXXXX.p8",
    keyId: "key-id",
    teamId: "developer-team-id"
  },
  production: true
};
```
However, after deploying my project to production on Vercel, the `.p8 key file` can't be found and I always get an error saying:
`VError: Failed loading token key: ENOENT: no such file or directory, open '/apn/AuthKey.p8'`

I followed this tutorial:
https://medium.com/@boris.poehland.business/next-js-api-routes-how-to-read-files-from-directory-compatible-with-vercel-5fb5837694b9
when trying to achieve it but with no success.

My code inside `/pages/api/sendPushNotification.ts` looks like this:

```
const apn = require("apn");
import fs from "fs";
import path from "path";

export default async (req, res) => {
  const dirRelativeToPublicFolder = "apn";
  const dir = path.resolve("./public", dirRelativeToPublicFolder);
  const filename = fs.readdirSync(dir);

  const authKey = path.join("/", dirRelativeToPublicFolder, filename[0]);

  const { iosTokens } = req.body;

  res.statusCode = 200;
  res.json({
    message: "Notification sent!",
  });

  if (iosTokens?.length !== 0) {
    const apnOptions = {
      token: {
        key: authKey,
        keyId: process.env.NEXT_PUBLIC_APN_KEY_ID,
        teamId: process.env.NEXT_PUBLIC_APN_TEAM_ID,
      },
      production: true,
    };

    const apnProvider = new apn.Provider(apnOptions);

    const note = new apn.Notification();
    note.alert = {
      title: "XXXXX",
      body: "XXXXXXXXXX",
    };
    note.sound = "default";
    note.topic = "XXXXXXX";

    console.log(
      `Attempting to send the notification to ${iosTokens.length} iOS devices.`
    );

    iosTokens.forEach((deviceToken) => {
      apnProvider
        .send(note, deviceToken)
        .then((result) => {
          console.log(result);
        })
        .catch((error) => {
          console.log(
            "Error occured when trying to send notification: ",
            error
          );
        });
    });
  }
}
```

**Comment by leerob** [comment, marked as answer]
https://github.com/vercel/next.js/issues/8251
